gpkit/substitution.py

```diff
@@ -179,8 +178,7 @@ def substitution(nomial, substitutions, val=None):
                 sub = VarKey(name=sub, **descr)
                 exps_[i] += HashVector({sub: x})
                 varlocs_[sub].append(i)
-            elif isinstance(sub, VarKey):
-                sub = VarKey(sub)
+            elif isinstance(sub, (VarKey, Monomial)):
                 if isinstance(var.units, Quantity):
                     try:
                         new_units = var.units/sub.units
```

Comment:
Yeah, I mostly gave up and flipped the switch so that it worked properly
for a test I wrote for the refactor...
On Mar 7, 2016 21:18, "Warren Hoburg" <notifications@github.com> wrote:

> In gpkit/substitution.py
> <https://github.com/hoburg/gpkit/pull/504#discussion_r55307673>:
>
> > @@ -179,8 +178,7 @@ def substitution(nomial, substitutions, val=None):
> >                  sub = VarKey(name=sub, **descr)
> >                  exps_[i] += HashVector({sub: x})
> >                  varlocs_[sub].append(i)
> > -            elif isinstance(sub, VarKey):
> > -                sub = VarKey(sub)
> > +            elif isinstance(sub, (VarKey, Monomial)):
> >                  if isinstance(var.units, Quantity):
> >                      try:
> >                          new_units = var.units/sub.units
>
> argh, I'm struggling with this.
>
> in your example above, if a = 1 [m], and y is in [cm], say 5 cm, then 100
> y is 500... [m]? that can't be correct.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hoburg/gpkit/pull/504/files#r55307673>.
>